src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultAssemblyPartDiscoveryProvider.cs

```diff
@@ -74,7 +74,7 @@ private class CandidatesResolver
             public CandidatesResolver(IReadOnlyList<RuntimeLibrary> dependencies, ISet<string> referenceAssemblies)
             {
                 _dependencies = dependencies
-                    .ToDictionary(d => d.Name, d => CreateDependency(d, referenceAssemblies));
+                    .ToDictionary(d => d.Name, d => CreateDependency(d, referenceAssemblies), StringComparer.OrdinalIgnoreCase);
```

Comment:
This means the candidate assemblies are case insensitive too right? like `Foo` and `foo` should match?...if yes, can you add a `GetLibrary(..` thingy for this case